DataOnion/Auth/AuthServiceStrategy.cs

```diff
@@ -0,0 +1,181 @@
+using StackExchange.Redis;
+using Microsoft.Extensions.Logging;
+
+namespace DataOnion.Auth
+{
+    internal interface IAuthServiceStrategy<T>
+    {
+        Task<T?> LoginAsync(T userSession);
+        Task<T?> GetSessionAsync(string id);
+        Task LogoutAsync(string id);
+    }
+
+    public interface IAuthStorable<T> : IEquatable<T>
+    {
+        string GetId();
+        IEnumerable<HashEntry> ToRedisHash();
+    }
+
+    public class SlidingExpirationConfig<T>
+    {
+        public TimeSpan SlidingExpiration { get; set; }
+        public TimeSpan? AbsoluteExpiration { get; set; }
+        public string ExpirationKey { get; set; }
+        public string EnvironmentPrefix { get; set; }
+        public string AuthPrefix { get; set; }
+        public Func<HashEntry[], T> ConstructFromHash { get; set; }
+    }
+
+    public class SlidingExpirationStrategy<T> : IAuthServiceStrategy<T>
+        where T : class, IAuthStorable<T>
```

Comment:
im thinking that i changed my mind in that the interface should be public but the implementation is internal

that way we can add a path in the fluent configuration API for consumers registering a custom auth strategy

that way in the case we need a different strategy later for another codebase we can just implement it rather than waiting for an update to the library